library/core/src/ptr/mod.rs

```diff
@@ -954,6 +954,23 @@ pub const unsafe fn swap<T>(x: *mut T, y: *mut T) {
     }
 }
 
+/// Non-overlapping *typed* swap of a single value.
```

Comment:
Should it be called `typed_swap_nonoverlapping`?

Reply:
I feel like the "nonoverlapping" distinction is way more fundamental than the "one" distinction. And since it is internal, consistency with `ptr::swap_nonoverlapping` isn't very important IMO.